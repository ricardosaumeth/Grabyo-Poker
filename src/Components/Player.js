import React, { Component } from 'react';
import { Card, PlayerHand, Button } from '../Styles/Styled';
import Input from './Common/Input';

export default class Players extends Component {
  constructor(props) {
    super(props);

    this.state = {
      isInputDisabled: true
    };

    this.textInput = React.createRef();
    this.onHandlerEditBtn = this.onHandlerEditBtn.bind(this);
  }

  onHandlerEditBtn() {
    this.setState({ isInputDisabled: false });
    this.textInput.current.textInput.current.focus();
  }

  renderPlayerHand() {
    const { playerHand } = this.props;
    if (playerHand && Array.isArray(playerHand)) {
      let cards = [];
      for (let i = 0; i < 5; i++) {
        cards.push(
          <Card key={i} suit={playerHand[i].suit} value={playerHand[i].value}>
            {playerHand[i].value}
          </Card>
        );
      }

      return <PlayerHand>{cards}</PlayerHand>;
    } else {
      return;
    }
  }

  render() {
    const { playerName, removed } = this.props;
    const { isInputDisabled } = this.state;
    return (
      <article>
        <p>
          <Input
            playerName={playerName}
            isDisabled={isInputDisabled}
            ref={this.textInput}
          />
          <Button onClick={this.onHandlerEditBtn}>
            <span role="img" alt="pencil" aria-label="pencil">
              ‚úèÔ∏è
            </span>
            Edit
          </Button>
          <Button onClick={removed}>
            <span role="img" alt="flame" aria-label="flame">
              üî•
            </span>
            Remove
          </Button>
        </p>
        {this.renderPlayerHand()}
      </article>
    );
  }
}
