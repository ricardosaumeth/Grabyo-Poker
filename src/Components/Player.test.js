import renderer from 'react-test-renderer';
import React from "react";
import "../setupTests"

import Player from "./Player";

describe(`Card deck`, () => {
	test('should match the Snapshot', () => {
    const playerHand = [
      {name: "A of S", suit: "S", value: "A", isSelected: true},
      {name: "8 of C", suit: "C", value: "8", isSelected: true},
      {name: "4 of S", suit: "S", value: "4", isSelected: true},
      {name: "8 of D", suit: "D", value: "8", isSelected: true},
      {name: "9 of H", suit: "H", value: "9", isSelected: true}
    ];
		const wrapper = renderer.create(
      <Player playerName='Player 1' 
        key='1'
        remopved={() => { }}
        playerHand={playerHand}
      />).toJSON();
		expect(wrapper).toMatchSnapshot();
	});
	
});