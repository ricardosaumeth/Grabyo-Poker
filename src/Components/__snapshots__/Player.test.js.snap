// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card deck should match the Snapshot 1`] = `
<article>
  <p>
    <input
      disabled={true}
      onChange={[Function]}
      placeholder="Player 1"
      type="text"
      value=""
    />
    <button
      className="sc-ifAKCX bKJSRJ"
      onClick={[Function]}
    >
      <span
        alt="pencil"
        aria-label="pencil"
        role="img"
      >
        ‚úèÔ∏è
      </span>
      Edit
    </button>
    <button
      className="sc-ifAKCX bKJSRJ"
    >
      <span
        alt="flame"
        aria-label="flame"
        role="img"
      >
        üî•
      </span>
      Remove
    </button>
  </p>
  <div
    className="sc-EHOje gsupyp"
  >
    <div
      className="sc-htpNat egytKp"
      value="A"
    >
      A
    </div>
    <div
      className="sc-htpNat bgQLPz"
      value="8"
    >
      8
    </div>
    <div
      className="sc-htpNat egytKp"
      value="4"
    >
      4
    </div>
    <div
      className="sc-htpNat lmaxQV"
      value="8"
    >
      8
    </div>
    <div
      className="sc-htpNat jXZeqR"
      value="9"
    >
      9
    </div>
  </div>
</article>
`;
